### PR TITLE
Remove discreteCNAData from ResultsViewPageStore

### PR DESCRIPTION
**Pull request by adamabeshouse**

### CODE REVIEW

File: src/shared/lib/StoreUtils.ts [hunk 1]
```diff
@@ -465,6 +465,30 @@ export async function fetchCnaOncoKbData(uniqueSampleKeyToTumorType:{[uniqueSamp
...
+                                          molecularProfileIdToMolecularProfile:{[molecularProfileId:string]:MolecularProfile},
+                                         client: OncoKbAPI = oncokbClient)
+{
+    if (!geneMolecularData.result || geneMolecularData.result.length === 0) {
+        return ONCOKB_DEFAULT;
+    }
+    else
+    {
+        const alterationsToQuery = _.filter(geneMolecularData.result, d=>{
+            return molecularProfileIdToMolecularProfile[d.molecularProfileId].molecularAlterationType === "COPY_NUMBER_ALTERATION" &&
```
Comment (alisman) [inline src/shared/lib/StoreUtils.ts]: There's a constants collection for this AlterationTypeConstants

File: src/shared/lib/StoreUtils.ts [hunk 2]
```diff
@@ -465,6 +465,30 @@ export async function fetchCnaOncoKbData(uniqueSampleKeyToTumorType:{[uniqueSamp
...
+                                         geneMolecularData:MobxPromise<GeneMolecularData[]>,
+                                          molecularProfileIdToMolecularProfile:{[molecularProfileId:string]:MolecularProfile},
+                                         client: OncoKbAPI = oncokbClient)
+{
+    if (!geneMolecularData.result || geneMolecularData.result.length === 0) {
+        return ONCOKB_DEFAULT;
+    }
+    else
+    {
+        const alterationsToQuery = _.filter(geneMolecularData.result, d=>{
```
Comment (alisman) [inline src/shared/lib/StoreUtils.ts]: i think d is MolecularData. we shoudl just name is molecularData or some such.  so much easier to read

File: src/shared/lib/StoreUtils.ts [hunk 3]
```diff
@@ -465,6 +465,30 @@ export async function fetchCnaOncoKbData(uniqueSampleKeyToTumorType:{[uniqueSamp
...
+            return molecularProfileIdToMolecularProfile[d.molecularProfileId].molecularAlterationType === "COPY_NUMBER_ALTERATION" &&
+                !!annotatedGenes[d.entrezGeneId];
+        });
+        const queryVariants = _.uniqBy(_.map(alterationsToQuery, (datum: GeneMolecularData) => {
+            return generateQueryVariant(datum.entrezGeneId,
+                cancerTypeForOncoKb(datum.uniqueSampleKey, uniqueSampleKeyToTumorType),
+                getAlterationString(parseInt(datum.value, 10)));
+        }), "id");
```
Comment (alisman) [inline src/shared/lib/StoreUtils.ts]: as nice as it is to use the string shortcut, we should use (molduclarData)=>molecularData.id  
to get the typescript safety.   We should just establish that habit.

Comment (adamabeshouse) [reply src/shared/lib/StoreUtils.ts]: I just copied this over but you're right

File: src/shared/lib/StoreUtils.ts (hunk 2 above)
Comment (adamabeshouse) [reply src/shared/lib/StoreUtils.ts]: sure

File: src/shared/lib/StoreUtils.ts (hunk 1 above)
Comment (adamabeshouse) [reply src/shared/lib/StoreUtils.ts]: right

**Pull request merged**